### PR TITLE
Fixes #7830 - interfaces api output is class specific

### PR DESCRIPTION
**Pull request by tstrachota**
- per type rabl templates
- fixed api docs for interfaces

### CODE REVIEW

File: app/controllers/api/v2/interfaces_controller.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require_dependency File.join(Rails.root, "app/models/nic/base")
```
Comment (ares) [inline app/controllers/api/v2/interfaces_controller.rb:1]: why do we need this?

File: app/views/api/v2/interfaces/main.json.rabl [hunk 2]
```diff
@@ -2,5 +2,9 @@ object @interface
 
 extends "api/v2/interfaces/base"
 
-attributes :username, :password, :provider, :subnet_id, :subnet_name, :domain_id, :domain_name, :created_at, :updated_at,
-           :managed, :virtual, :identifier, :tag, :attached_to, :mode, :attached_devices, :bond_options
+attributes :subnet_id, :subnet_name, :domain_id, :domain_name, :created_at, :updated_at,
+           :managed, :identifier
+
+node do |interface|
+  partial("api/v2/interfaces/types/#{interface.type_name.downcase}.json", :object => interface)
```
Comment (ares) [inline app/views/api/v2/interfaces/main.json.rabl:9]: this way we don't display managed attributes in children, e.g. in bond
currently it only applies to virtual attributes but I think we should render all inherited partials as well

Comment (tstrachota) [inline app/views/api/v2/interfaces/main.json.rabl:9]: You're right. I added `extends "managed"`to the templates for `BMC` and `Bond` interfaces to fix it.

File: app/controllers/api/v2/interfaces_controller.rb (hunk 1 above)
Comment (tstrachota) [inline app/controllers/api/v2/interfaces_controller.rb:1]: I need to load constants `Nic::TYPES`, `Nic::BMC::PROVIDERS` and `Nic::Bond::MODES` to use them in the apidoc param validators.

Comment (ares) [inline app/controllers/api/v2/interfaces_controller.rb:1]: I see, thanks

**Comment by ares** [comment]
Merged as 5da15d1ad20a71b65ad858d392e17fb46f074480, thanks @tstrachota!

**Pull request closed without merging**